### DISCUSSION TITLE
How do I cycle back to the first index?

### DISCUSSION THREAD

**Discussion by nyngwang** [dcampos/nvim-snippy#113] [Q&A] [answered]
For example, how do I cycle back to the `{1:condition}` after I'm at `{2:what}` with the following `.snippet`?

```
# Custom tabstop order
snippet repeat
	repeat
		${2:what}
	while ${1:condition}
```

**Comment by dcampos** [comment]
Because your tab stops are reversed, you should use your normal backward mapping to go to `${1}` if you're at `${2}`. For example, if you have `<tab>` mapped to jump forward and `<s-tab>` to move backward, press `<s-tab>` to go to `${1}` when you are at `${2}`.

Alternatively, you can use an ascending tab stop order to avoid the need for mental gymnastics.

**Reply by nyngwang** [reply]
Is it possible to support a one-way `<tab>` that can cycle them all? 1->2->3->1->(repeat) all with `<tab>`, that's what I want.

**Reply by dcampos** [reply, marked as answer]
No, that's not possible at the moment. You'd have to cycle back to the first stop using the backward jumping mapping.
